src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,1246 @@
+
+    return image
+
+
+def load_image_canonical(image: Union[torch.Tensor, np.ndarray, Image.Image]) -> Tuple[torch.Tensor, int]:
+    if isinstance(image, Image.Image):
+        image = np.array(image)
+
+    input_dtype_max = None
+    if isinstance(image, np.ndarray):
+        if np.issubdtype(image.dtype, np.integer) and not np.issubdtype(image.dtype, np.unsignedinteger):
+            raise ValueError(f"Input image dtype={image.dtype} cannot be a signed integer.")
+        if np.issubdtype(image.dtype, np.complexfloating):
+            raise ValueError(f"Input image dtype={image.dtype} cannot be complex.")
+        if np.issubdtype(image.dtype, bool):
+            raise ValueError(f"Input image dtype={image.dtype} cannot be boolean.")
+        if np.issubdtype(image.dtype, np.unsignedinteger):
+            input_dtype_max = np.iinfo(image.dtype).max
+            image = image.astype(np.float32)  # because torch does not have unsigned dtypes beyond torch.uint8
+        image = torch.from_numpy(image)
+
+    if torch.is_tensor(image) and not torch.is_floating_point(image) and input_dtype_max is None:
+        if image.dtype != torch.uint8:
+            raise ValueError(f"Image dtype={image.dtype} is not supported.")
+        input_dtype_max = 255
+
+    if image.dim() == 2:
+        image = image.unsqueeze(0).repeat(3, 1, 1)  # [3,H,W]
+    elif image.dim() == 3:
+        if image.shape[2] in (1, 3):
+            image = image.permute(2, 0, 1)  # [1|3,H,W]
+        if image.shape[0] == 1:
+            image = image.repeat(3, 1, 1)  # [3,H,W]
+        if image.shape[0] != 3:
+            raise ValueError(f"Input image is not 1- or 3-channel: {image.shape}.")
+    else:
+        raise ValueError("Input image is not a 2- or 3-dimensional tensor.")
+
+    image = image.unsqueeze(0)  # [1,3,H,W]
+
+    return image, input_dtype_max
+
+
+def check_image_values_range(image: torch.FloatTensor) -> None:
+    assert torch.is_floating_point(image)
+
+    val_min = image.min().item()
+    val_max = image.max().item()
+
+    if val_min < -1.0 or val_max > 1.0:
+        raise ValueError("Input image data is partially outside of the [-1,1] range.")
+    if val_min >= 0.0:
+        logger.warning(
+            "Input image data is entirely in the [0,1] range; expecting [-1,1]. "
+            "This could be an issue with normalization"
+        )
+
+
+def validate_vae(vae: torch.nn.Module) -> None:
+    if len(vae.config.down_block_types) != 4 or len(vae.config.up_block_types) != 4:
+        raise ValueError(f"Unexpected VAE down_block_types or up_block_types; {vae.config}")
+    if vae.config.in_channels != 3 or vae.config.out_channels != 3:
+        raise ValueError(f"Unexpected VAE input-output space size; {vae.config}")
+    if vae.config.latent_channels != 4:
+        raise ValueError(f"Unexpected VAE latent space size; {vae.config}")
+
+
+def find_batch_size(ensemble_size: int, resolution: int, dtype: torch.dtype) -> int:
```

Comment:
We let the user configure batch size instead of configuring it for them. It might be better to just have this information in the official pipeline documentation.

Reply:
Just a tiny note: the notion of batch_size is slightly different in these pipelines. We use it to perform batched inference of ensemble predictions for each image independently, and therefore the batch_size can be automatically detected from processing resolution. It may even be different for each input image. For example, if the user inputs a list of two images, one small and one large, this logic will apply a large batch_size for the smaller image and a smaller batch_size for the larger (unless batch_size is explicitly set).

Reply:
Fixed: default batch_size is now 1, and all implicit logic is now removed